clippy_lints/src/methods/mod.rs

```diff
@@ -1670,6 +1676,48 @@ impl<'tcx> LateLintPass<'tcx> for Methods {
             }
         }
     }
+
+    fn check_trait_item(&mut self, cx: &LateContext<'tcx>, item: &'tcx TraitItem<'_>) {
+        if_chain! {
+            if !in_external_macro(cx.tcx.sess, item.span);
+            if !item.span.from_expansion();
```

Comment:
I think `in_external_macro` would be redundant if we avoid code from expansion, but shouldn't suffice with avoiding external macros? Why should we avoid code from expansion in general?

Reply:
Good catch, I guess I forgot to remove these two lines after some local tests.
I agree we shouldn't avoid code from expansion.